zookeeper/tf/dataset.py

```diff
@@ -130,14 +132,14 @@ class MultiTFDSDataset(Dataset):
     # A non-empty mapping from dataset names as keys to splits as values. The
     # training data will be the concatenation of the datasets loaded from each
     # (key, value) pair.
-    train_split: Dict[str, str]
+    train_split: Dict[str, str] = Field()
 
     # As above, a mapping from dataset names as keys to splits as values. May be
     # empty, indicating no validation data.
-    validation_split: Dict[str, str] = {}
+    validation_split: Dict[str, str] = Field(lambda: {})
```

Comment:
What's up with the lambda here?

Reply:
Non-immutable objects have to be generated by a factory, in this case an argument-less lambda. Otherwise every instance of the class will receive the same field value instance, which is obviously problematic if the value is mutable.

`@dataclasses` prohibits mutable field values without a factory method for this reason.